src/daily_output.ts

```diff
@@ -0,0 +1,432 @@
+import { readFileSync, writeFileSync } from 'fs';
+import { GROUP_EFFORT_REWARDS, ONE_THOUSAND_MB, OUTPUT_NAME, OUTPUT_TEMPLATE_NAME } from './constants';
+import { GQLEdgeInterface } from './gql_types';
+import { OutputData, StakedPSTHolders, WalletsStats, WalletStatEntry } from './inferno_types';
+
+/**
+ * A class responsible of parsing the GQL and CommunityOracle data into the OutputData file
+ */
+export class DailyOutput {
+	private data = this.read();
+	private latestBlock = 0;
+	private latestTimestamp = 0;
+	private bundlesTips: { [address: string]: number } = {};
+
+	/**
+	 * Takes the data from the previously generated data, fallbacking to the base template if not present
+	 * @throws if the validation of the output file fails
+	 * @retuns {OutputData}
+	 */
+	public read(): OutputData {
+		const data = (() => {
+			try {
+				return this.readOutputFile();
+			} catch (err) {
```

Comment:
This try/catch is leaked all the way into the top scope here. I think we could handle this within `readOutputFile`

Reply:
Is the `readTemplate` just for development? Whats the use case for dummy information in production?

Reply:
readTemplate is used to instantiate a fresh OutputData, I could go for another approach